src/util/Interrupt.cpp

```diff
@@ -37,16 +41,23 @@ Interrupt::request()
     requested = true;
 }
 
+void
+Interrupt::requestForCurrentThread()
```

Comment:
I don't understand how that can work. If GEOS is busy in a thread T1 (processing thread), and you want to interrupt it, you need to do that from another thread T2 (GUI typically). But if in T2 you call Interrupt::requestForCurrentThread(), then it will set the thread_local requested_for_thread for T2, not T1.

Reply:
forget above comment... I figured it out